content/post/2025-06-19-open-files-folders-websites/index.Rmd

```diff
@@ -0,0 +1,154 @@
+Please tell me if you have any more information on this. :pray:
+
+In any case I really enjoy `file.edit()` in codebases.
+Interactively, I do not use is as often as, say, Positron's shortcut for navigating to files (Ctrl+P on my machine).
+
+Last but not least, if you want to make it easy for the user to open a file, without opening it on their behalf, in messages emitted through the [cli package](https://blog.r-hub.io/2023/11/30/cliff-notes-about-cli/) you can use the [file class](https://cli.r-lib.org/reference/inline-markup.html#classes):
+
+```{r, eval=FALSE}
+cli::cli_alert_success("Hey go edit {.file config.toml} please!")
+```
+> _"If the terminal supports ANSI hyperlinks (e.g. RStudio, iTerm2, etc.), then cli creates a clickable link that opens the file in RStudio or with the default app for the file type."_
+
+[^adventurous]: Look how adventurous I was when preparing this post! Running R from a terminal!
+[^atom]: Which is... [Atom](https://en.wikipedia.org/wiki/Atom_(text_editor)), meaning I live in the past since its end-of-life occurred a few years ago. :ghost: I actually rarely use it.
+
+## Open the script or test file of a script-test pair: `usethis::use_r()` and `usethis::use_test()`
+
+Coming back to the wonderful usethis package, when you name your test files after your scripts as you should,
+running [`usethis::use_test()`](https://r-pkgs.org/testing-basics.html#create-a-test) when focussed on an R script will open its test file, and conversely `usethis::use_r()` when focussed on a test file.
+This is still how I switch between the two even if inside Positron.
+
+## Open a project: `positron`, project launcher, {usethis}
+
+This is all very good when within a project, but how do you enter your IDE in the first place?
+How sad would it be to lose momentum before even launching a project?
+
+When I used RStudio IDE I would navigate to the folder of interest then double-click on the `.Rproj` file.
+Hannah Frick inspired me, when demoing Positron to me, to think a bit harder about how much I click. :sweat_smile:
+I have adopted a more keyboard-based workflow since I switched to Positron, which I could have done with RStudio IDE.
+
+One thing in particular that Hannah showed me that looks cool is the use of a [project launcher](https://positron.posit.co/rstudio-rproj-file.html#use-an-application-launcher).
+
+Now, I myself still use the terminal. 
+For instance if I clone a repository[^usethis], I'll then open it with the `positron` command.
+
+Another way to open projects I use a lot is, on Positron as on Rstudio, clicking to the recent projects -- when the IDE is already up and running.
+Unfortunately, I created and [currently](https://stateofther.netlify.app/post/saperlipopette/) [often](https://www.meetup.com/rbuenosaires/events/308338205/) use a [package that creates throwaway folders for practicing Git](https://docs.ropensci.org/saperlipopette/) which makes the list of recent projects utter rubbish. 🙃
```

Comment:
```suggestion
Unfortunately, I created and [currently](https://stateofther.netlify.app/post/saperlipopette/) [often](https://www.meetup.com/rbuenosaires/events/308338205/) use the [saperlipopette](https://docs.ropensci.org/saperlipopette/) package that creates throwaway folders for practicing Git.
This makes the list of recent projects utter rubbish. 🙃
```

Reply:
Otherwise, the package name in the next sentence might be surprising to readers who don't know about it yet and didn't follow the links above